app/serializers/lib_search_result_serializer.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class LibSearchResultSerializer < ObjectSerializer
+  set_id { |object| Fingerprintable::Fingerprinter.new(object: object).fingerprint }
+
+  attributes :pagination, :query
+
+  attribute :ledger do |_object, params|
+    params[:ledger].id
+  end
+
+  attribute :next_page_pagination do |object|
+    object.next_searcher.try(:pagination)
+  end
+
+  attribute :next_page_url do |object, params|
+    uri = URI.parse(params[:original_url])
+    uri_params = Rack::Utils.parse_nested_query(uri.query)
+    uri.query = uri_params.merge(
+      pagination: object.next_searcher.pagination
+    ).to_query
+    uri.to_s
+  end
+
+  attribute :previous_page_pagination do |object|
+    object.previous_searcher.try(:pagination)
+  end
+
+  attribute :previous_page_url do |object, params|
```

Comment:
I noticed that if there's no previous page, this URL will be the same as the one you queried the API with. I assume that conversely, if there's no next page, you'll get the same URL as the one you queried with. Do you think we should have that behavior? Or instead, we could return `null` which would indicate that there is no next page

Reply:
Yes, totally should be the case.  Will fix.